### PR TITLE
feat: add custom merge option

### PR DESCRIPTION
**Pull request by alanblins**
Hi,
This is extension to allow the library use custom merge function. By default the library `jsonMerge` uses `Object.assign` that has some limitations. Example:
Consider the object below:
```
var object = {
  'a': [{ 'b': 2 }, { 'd': 4 }]
};
 
var other = {
  'a': [{ 'c': 3 }, { 'e': 5 }]
};
```
The command `Object.assign(object, other)` won't merge the keys `b` and `d`.
```
{
 "a":
   [
     {"c":3},
     {"e":5}
   ]
}
```

The [lodash.merge](https://lodash.com/docs/4.17.15#merge) function solves that problem:
```
const _ = require('lodash');
_.merge(object. other);
```
Output
```
{ 'a': [{ 'b': 2, 'c': 3 }, { 'd': 4, 'e': 5 }] }
```

### CODE REVIEW

File: __tests__/index.test.js [hunk 1]
```diff
@@ -4,6 +4,7 @@ const path = require('path');
 const { build } = require('esbuild');
 const fse = require('fs-extra');
 
+const defaultComposer = require('./defaultComposer');
```
Comment (karolis-sh) [inline __tests__/index.test.js:7]: This function is rather complex and pushes towards the case when one would need a test for a test-only code, which isn't ideal. I assume this is a copy paste from lodash? I think it's fine to add it as a dev dependency and use it directly, that would make tests simpler.

**Review by karolis-sh — CHANGES_REQUESTED** [review]

**Comment by karolis-sh** [comment]
Hi, yeah legit feature, thanks for the PR!

File: __tests__/index.test.js (hunk 1 above)
Comment (alanblins) [reply __tests__/index.test.js]: Thanks for the feedback. This function is a copy past from the library [default-composer](https://www.npmjs.com/package/default-composer). I updated the pull request and I added it on dev dependency.  I didn't added lodash as it is 1.41mb. This default-composer library is only 41kb.

Comment (karolis-sh) [reply __tests__/index.test.js]: The size mostly doesn't matter for dev deps, especially as they won't be installed when consuming this package. It's more about understating if the function provided was used correctly in the test (so either mock, something well - known or extremely simple). Anyway, let's move on

**Pull request merged**